pyOCD/gdbserver/gdbserver.py

```diff
@@ -24,10 +24,185 @@
 import sys
 from gdb_socket import GDBSocket
 from gdb_websocket import GDBWebSocket
+import traceback
+import Queue
+
+CTRL_C = '\x03'
 
 # Logging options. Set to True to enable.
 LOG_MEM = False # Log memory accesses.
 LOG_ACK = False # Log ack or nak.
+LOG_PACKETS = False # Log all packets sent and received.
+
+def checksum(data):
+    return "%02x" % (sum([ord(c) for c in data]) % 256)
+
+## @brief Exception used to signal the GDB server connection closed.
+class ConnectionClosedException(Exception):
+    pass
+
+## @brief Packet I/O thread.
+#
+# This class is a thread used by the GDBServer class to perform all RSP packet I/O. It
+# handles verifying checksums, acking, and receiving Ctrl-C interrupts. There is a queue
+# for received packets. The interface to this queue is the receive() method. The send()
+# method writes outgoing packets to the socket immediately.
+class GDBServerPacketIOThread(threading.Thread):
+    def __init__(self, abstract_socket):
+        super(GDBServerPacketIOThread, self).__init__(name="gdb-packet-thread")
+        self._abstract_socket = abstract_socket
+        self._receive_queue = Queue.Queue()
+        self._shutdown_event = threading.Event()
+        self.interrupt_event = threading.Event()
+        self.send_acks = True
+        self._clear_send_acks = False
+        self._buffer = ''
+        self._expecting_ack = False
+        self.time_of_last_packet = 0
```

Comment:
It looks like time_of_last_packet is only set and never read.  Are there plans for it, or should it be removed?